lib/nxt_support/util/crystalizer.rb

```diff
@@ -0,0 +1,34 @@
+module NxtSupport
+  class Crystallizer
+    Error = Class.new(StandardError)
+
+    include NxtInit
+
+    attr_init :collection, on_uniqueness_violation: -> { uniqueness_violation_handler }, attribute: :itself
+
+    def call
+      ensure_values_are_unique
```

Comment:
To me `ensure_values_are_unique` implies that each value appears only once in the collection (every value is distinct), but we're actually checking for sameness. Maybe `ensure_values_are_same` or `ensure_equal_values`?

Reply:
Additionally it validates that there is exactly one (and not zero) items in the collection. Maybe call it completely abstract, e.g. `ensure_values_are_crystallizeable`?

Reply:
- `ensure_unanimity!`
- `unanimous!`